### DISCUSSION TITLE
Prevent relation model creation when using make(One) on model factories

### DISCUSSION THREAD

**Discussion by renegeuze** [laravel/framework#50985] [Ideas]
Model factories contain the methods `make` and `makeOne` which do generate a model without creating a database record.
However, when the definition contains a relation to another factory. It always defaults to using `create` for the relation.
If often results in accidentally slowing down tests by creating unnecessary database records.

Before I create any Pull request with the required changes, I would like some input. And maybe I've missed some existing way to not have this issue at all.

Example definition for "TenderFactory":
```
public function definition(): array
{
    return [
        'title' => $this->faker->sentence,
        'client_id' => ClientFactory::new(),
        'status' => TenderStatus::DRAFT,
    ];
}
```

One way to fix it is by writing `ClientFactory::new()->makeOne()`, but then using `create` on the TenderFactory will break.

My suggestion would be to switch "mode" from `create` to `make` for the relations when `make` or `makeOne` is called.

For our own small project, I've already made a working proof of concept using the following changes:
In **src/Illuminate/Database/Eloquent/Factories/Factory.php**
```
protected bool $allowCreatingRelations;

public function allowCreatingRelations(bool $allowCreatingRelations = true): self
{
    $this->allowCreatingRelations = $allowCreatingRelations;

    return $this;
}

public function make($attributes = [], ?Model $parent = null)
{
    $this->allowCreatingRelations ??= false;
    ... existing code
}

public function create($attributes = [], ?Model $parent = null)
{
    $this->allowCreatingRelations ??= true;
    ... existing code
}

protected function expandAttributes(array $definition)
{
    ... existing code
            $mode = ($this->allowCreatingRelations ?? false) ? 'create' : 'make';
            
            $attribute = $this->getRandomRecycledModel($attribute->modelName())?->getKey()
                ?? $attribute->recycle($this->recycle)->$mode()->getKey();

    ... existing code
}
```

Some questions and notes I have
* Useful to change? Or is this just a me problem.
* Any general improvements in code?
* Any unforeseen side effects?
* This would be a breaking change for test suites that (imo incorrectly) make use of the created records.

**Comment by finpension-aless** [comment]
@renegeuze Just came across this older post because I have a similar issue. Did you notice the variable $expandRelationships (or the static $expandRelationshipsByDefault) int the abstract Factory class? I just tested it briefly, but I think that's what you are looking for.

**Comment by shaedrich** [comment]
Yeah, that bugs me, too. This makes it impossible to use PHPUnit's `TestCase` class with Factories when they have relations, which adds a lot of overhead for simple unit tests that wouldn't need the whole framework to be booted.

However, in #58277, it was attempted to find a solution for this, but the PR was rejected.
